File: movie-client/src/app/movie.ts
import { Rating } from "./rating";

export class Movie {
    public id: number = 0;
    public name: String = '';
    public description: String = '';
    public rating: Rating = new Rating();
    public cover: String = '';

    Constructor(id: number, name: String, description: String, cover: String){
        this.id = id;
        this.name=name;
        this.description=description;
        this.cover=cover;
    }

}

File: movie-info-service/src/main/java/io/javabrains/movieinfoservice/Movie.java
package io.javabrains.movieinfoservice;

import java.util.Objects;
import javax.persistence.Column;
import javax.persistence.GenerationType;
import javax.persistence.GeneratedValue;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;

import org.springframework.hateoas.server.core.Relation;

@Relation(collectionRelation="data")
@Entity
@Table(name = "MOVIES")
public class Movie {

    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    @Column(name="ID")
    private int id;

    @Column(name="NAME")
    private String name;

    @Column(name="DESCRIPTION")
    private String description;

    public Movie() {

    }

    public Movie(int id, String name, String description) {
        setId(id);
        setName(name);
        setDescription(description);
    }

    public int getId() {
        return id;
    }
    public void setId(int id) {
        this.id = id;
    }
    public String getName() {
        return name;
    }
    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}
